src/Components/TestExplorer.fs

```diff
@@ -131,59 +131,91 @@ module DotnetTest =
-                      ErrorStackTrace = errorStackTrace
-                      Expected = expected
-                      Actual = actual
-                      Timing = timing })
+                                $"/t:TestRun/t:Results/t:UnitTestResult[@executionId='{executionId}']/@outcome"
+
+                        let errorInfoMessage =
+                            xpathSelector.TrySelectString
+                                $"/t:TestRun/t:Results/t:UnitTestResult[@executionId='{executionId}']/t:Output/t:ErrorInfo/t:Message"
+
+                        let errorStackTrace =
+                            xpathSelector.TrySelectString
+                                $"/t:TestRun/t:Results/t:UnitTestResult[@executionId='{executionId}']/t:Output/t:ErrorInfo/t:StackTrace"
+
+                        let timing =
+                            let duration =
+                                xpathSelector.SelectString
+                                    $"/t:TestRun/t:Results/t:UnitTestResult[@executionId='{executionId}']/@duration"
+
+                            TimeSpan.Parse(duration).TotalMilliseconds
+
+                        let expected, actual =
+                            match errorInfoMessage with
+                            | None -> None, None
+                            | Some message ->
+                                let lines =
+                                    message.Split([| "\r\n"; "\n" |], StringSplitOptions.RemoveEmptyEntries)
+                                    |> Array.map (fun n -> n.TrimStart())
+
+                                let tryFind (startsWith: string) =
+                                    Array.tryFind (fun (line: string) -> line.StartsWith(startsWith)) lines
+                                    |> Option.map (fun line -> line.Replace(startsWith, "").TrimStart())
+
+                                tryFind "Expected:", tryFind "But was:"
+
+                        if Seq.length testCases > 1 then
+                            let ti =
+                                t.Test.children.get (t.Test.uri.Value.ToString() + " -- " + testName)
+                                |> Option.defaultWith (fun () ->
+                                    tc.createTestItem (
+                                        t.Test.uri.Value.ToString() + " -- " + testName,
+                                        testName,
+                                        t.Test.uri.Value
+                                    ))
+
+                            t.Test.children.add ti
+
+                        { Test = t.Test
+                          FullTestName = fullName
+                          Outcome = !!outcome
+                          ErrorMessage = errorInfoMessage
+                          ErrorStackTrace = errorStackTrace
+                          Expected = expected
+                          Actual = actual
+                          Timing = timing }))
+                |> Array.concat
```

Comment:
nit: normally a map + concat can be rewritten to use `collect` instead - can you check into that?